src/IO/HDF5/HDF5IOHandler.cpp

```diff
@@ -1588,8 +1588,8 @@ void HDF5IOHandlerImpl::writeAttribute(
         auto vs = att.get<std::vector<std::string> >();
         size_t max_len = 0;
         for (std::string const &s : vs)
-            max_len = std::max(max_len, s.size());
-        std::unique_ptr<char[]> c_str(new char[max_len * vs.size()]);
+            max_len = std::max(max_len, s.size() + 1);
+        std::unique_ptr<char[]> c_str(new char[max_len * vs.size()]());
```

Comment:
Subtle bug fixed in HDF5IOHandler writeAttribute() method:
`new char[…]` does not initialize the memory, but `strncopy` will copy until either a NULL is found or the `max_len` has been reached, so the uninitialized memory is always guarded from actually being read. Also, no segfault is triggered, since the memory left and right of the uninitialized memory is written and the virtual memory can detect no illegal access in that case.

Before this PR, a `std::vector{"VECTOR", "OF", "STRINGS"}` was converted to the following `c_str` buffer:
```
VECTOR0
OF0****
STRINGS
```
where 0 is a null terminator and * is uninitialized memory.
With this PR, it is converted to:
```
VECTOR00
OF000000
STRINGS0
```
Every line has at least one null terminator, and the rest of the line is also padded with nulls.

Reply:
Awesome, thanks a lot for the fix and explanation.